src/common/kmer_index/extension_index/inout_mask.hpp

```diff
@@ -63,8 +63,8 @@ class InOutMask {
         return unique[mask];
     }
 
-    static constexpr char GetUnique(uint8_t mask) {
-        constexpr char next[] =
+    static constexpr int8_t GetUnique(uint8_t mask) {
```

Comment:
Thanks for the change! However, I think we need to go the other way. Here is the full story:
 - The return value is assumed to be `0 - 3` (representing `A`, `C`, `G`, `T` correspondingly) for input of `0`, `1`, `2, `4`.
 - `-1`'s shows the impossible values, but still they should be as initializers.

So, the return type should really be `char` as it is used later on in `GetUniqueOutgoing` / `GetUniqueIncoming`.

So, I would probably use `0xFF` instead of `-1`. Does this work for you on aarch64-linux?

Reply:
Yes! That works too!
Let me update the PR!

Reply:
I updated it but I have the feeling that more changes are needed at the call sites of `GetUniqueOutgoing` / `GetUniqueIncoming`.

Reply:
Callsites should be fine. The contract is that one should never call `GetUniqueOutgoing` / `GetUniqueIncoming` if they could return `0xFF`.